huaweicloud/services/cce/resource_huaweicloud_cce_cluster.go

```diff
@@ -1074,6 +1095,47 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, meta int
 	return resourceClusterRead(ctx, d, meta)
 }
 
+func buildUpdateClusterConfigurationsBodyParams(d *schema.ResourceData) (map[string]interface{}, error) {
+	configurationsPackages, err := buildConfigurationsPackagesBodyParams(d)
+	if err != nil {
+		return nil, err
+	}
+	bodyParams := map[string]interface{}{
+		"apiVersion": "v3",
+		"kind":       "Configuration",
+		"metadata": map[string]interface{}{
+			"name": "configuration",
+		},
+		"spec": map[string]interface{}{
+			"packages": configurationsPackages,
+		},
+	}
+	return bodyParams, nil
+}
+
+func buildConfigurationsPackagesBodyParams(d *schema.ResourceData) ([]map[string]interface{}, error) {
+	packagesRaw := d.Get("component_configurations").([]interface{})
+	bodyParams := make([]map[string]interface{}, len(packagesRaw))
+	for i, v := range packagesRaw {
+		packageRaw := v.(map[string]interface{})
+		bodyParams[i] = map[string]interface{}{
+			"name": packageRaw["name"],
+		}
+
+		var configurations interface{}
```

Comment:
move it to if statement? it is only used in the if statement

Reply:
sure